### PR TITLE
feat(infra): add ability to mount nodes to nearest FrameworkScope root

### PR DESCRIPTION
**Pull request by CatsJuice**

### CODE REVIEW

**Comment by CatsJuice** [comment]
* **#7593** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7593?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7551** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7551?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7587** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7587?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#7578](https://github.com/toeverything/AFFiNE/pull/7578) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7578?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CatsJuice and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by forehalo — APPROVED** [review]

File: packages/frontend/core/src/components/page-list/view/edit-collection/hooks.tsx [hunk 1]
```diff
@@ -15,8 +16,11 @@ export const useSelectPage = ({
   const close = useCallback(() => {
     onChange(undefined);
   }, []);
-  return {
-    node: (
+
+  const { add } = useScopeRootComponents('select-page-modal');
```
Comment (EYHN) [inline packages/frontend/core/src/components/page-list/view/edit-collection/hooks.tsx]:
```suggestion
  const { mount } = useMount('select-page-modal');
```

File: packages/common/infra/src/framework/react/index.tsx [hunk 2]
```diff
@@ -126,7 +129,7 @@ export const FrameworkScope = ({
 
   return (
     <FrameworkStackContext.Provider value={nextStack}>
-      {children}
+      <ScopeRootComponents>{children}</ScopeRootComponents>
```
Comment (EYHN) [inline packages/common/infra/src/framework/react/index.tsx]:
```suggestion
      <MountPoint>{children}</MountPoint>
```

**Review by EYHN — COMMENTED** [review]
Maybe a different name would be better?

File: packages/frontend/core/src/components/page-list/view/edit-collection/hooks.tsx (hunk 1 above)
Comment (CatsJuice) [reply packages/frontend/core/src/components/page-list/view/edit-collection/hooks.tsx]: `useMount` seems a little vague, how about ﻿`useMountToNearestScope`?

Comment (EYHN) [reply packages/frontend/core/src/components/page-list/view/edit-collection/hooks.tsx]: I think we'd better separate this from `FrameworkScope`. Sometimes other contexts also want to nest a mountpoint, so it's better not to include `scope` in the name.

**Pull request merged**